packages/nuxt/src/app/components/nuxt-link.ts

```diff
@@ -314,12 +362,11 @@ export function defineNuxtLink (options: NuxtLinkOptions) {
               if (!href) { return undefined }
 
               const url = parseURL(href)
-              return {
+              return <RouteLocation & { href: string }> {
```

Comment:
maybe we could use `satisfies` which would ensure we actually do return values compatible with the vue router types?